_pages/dev/ssr/server-side-rendering-coding-guidelines.md

```diff
@@ -57,7 +57,11 @@ constructor(
 
 **Note:** The `@Optional()` decorator is necessary. If it is not included, the injection will crash for client-side rendering (CSR) because these tokens are not provided in CSR.
 
-### Workaround for Known Issue in Spartacus 3.0.2 and Earlier
+## Avoiding Memory Leaks in SSR
+
+In the SSR server, there is one long-living Node.js process that handles all HTTP requests. On each request, this process bootstraps the Angular application, returns the HTML to the client, and cleans up the application's resources. If any subscription that was created in any Angular (singleton) service is not disposed of on app destroy, this subscription remains pending, even after the app is destroyed, and this causes a memory leak. In SSR, Angular calls `ngOnDestroy` for services, so it is a good place to unsubscribe any pending RxJs subscriptions at the end of life of the service.
```

Comment:
```suggestion
In the SSR server, there is one long-living Node.js process that handles all HTTP requests. On each request, this process bootstraps the Angular application, returns the HTML to the client, and cleans up the application's resources. If any subscription that was created in any Angular (singleton) service is not disposed on app destroy, this subscription remains pending, even after the app is destroyed, and this causes a memory leak. In SSR, Angular calls `ngOnDestroy` for services, so it is a good place to unsubscribe any pending RxJs subscriptions at the end of life of the service.
```

I'm not native English speaker, but I feel the word 'of' is a typo there. Correct?

Reply:
In fact, the "of" is necessary 🙂 

One does not "dispose something", but rather, one "disposes *of* something". I can easily imagine that it might seem strange to add it, but in fact, to a native English speaker, it sounds strange without it 🙂 

Thank you for raising this, though!